### PR TITLE
Support Windows

### PR DESCRIPTION
**Pull request by thxCode**
- Use to deploy nginx-proxy
- Use to deploy network plugin
- Use to deploy kubelet, kube-proxy
- Support 1903

**Issue:**
https://github.com/rancher/rancher/issues/16460

### CODE REVIEW

File: windows/cloud-provider.psm1 [hunk 1]
```diff
@@ -0,0 +1,139 @@
+$ErrorActionPreference = 'Stop'
+$WarningPreference = 'SilentlyContinue'
+$VerbosePreference = 'SilentlyContinue'
+$DebugPreference = 'SilentlyContinue'
+$InformationPreference = 'SilentlyContinue'
+
+function Log-Info
```
Comment (orangedeng) [inline windows/cloud-provider.psm1]: seems like logs' function is in every powershell scripts. Can we move them into one file?

File: windows/sidecar.ps1 [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+                $vni = $options.flannel_backend_vni
+                $port = $options.flannel_backend_port
+
+                if (-not $type) {
+                    $type = "vxlan"
+                }
+                if (-not $vni) {
+                    $vni = 4096
+                } else {
+                    $vni = [int]$vni
```
Comment (orangedeng) [inline windows/sidecar.ps1]: We should comment that windows doesn't support changing vni for now.

File: windows/cloud-provider.psm1 (hunk 1 above)
Comment (thxCode) [reply windows/cloud-provider.psm1]: Refactored, going to test again.

File: windows/sidecar.ps1 (hunk 2 above)
Comment (thxCode) [reply windows/sidecar.ps1]: Commented

**Comment by orangedeng** [comment]
LGTM

**Comment by loganhz** [comment]
There is no manifest.tmpl in your PR

**Comment by thxCode** [comment]
Updated

**Pull request merged**